lib/dialogs/import/importers.ts

```diff
@@ -26,4 +28,46 @@ const importers: Array<importer> = [
   },
 ];
 
-export default importers;
+export const getImporters = () => {
+  return importers;
+};
+
+export const forFilename = (file: String): Importer => {
+  const fileExtension =
+    file.substring(file.lastIndexOf('.') + 1, file.length) || file;
+
+  switch (fileExtension) {
+    case 'json':
+      return importers.find((importer) => importer.name === 'simplenote');
+
+    case 'enex':
+      return importers.find((importer) => importer.name === 'evernote');
+
+    case 'txt':
+    case 'md':
+      return importers.find((importer) => importer.name === 'text-files');
+  }
```

Comment:
given that there's only a few possibilities we shouldn't need to find. `importers` is right above in this same module

```ts
switch (fileExtension) {
	case 'json':
		return simplenoteImporter;

	case 'enex':
		return evernoteImporter;

	case 'txt':
	case 'md':
		return textfilesImporter;
}
```

this may demand that we extract the individual importer definitions from the list…

```ts
const simplenoteImporter: Importer { … }
…

const importers = [
	simplenoteImporter,
	evernoteImporter,
	textfilesImporter
];
```

Reply:
Sounds good. I'll update this!